lib/action_view/component/base.rb

```diff
@@ -164,7 +164,7 @@ def type
         end
 
         def identifier
-          ""
+          source_location
```

Comment:
But this is the source location of the component class, not the html itself no? 

And you would need to it per variant I think 🤔

Reply:
You're 100% right. I overlooked the `source_location` path. We have the template path under `templates.first[:path]`. However, I'm not sure how we'd handle for multiple templates.

Also, how would we take the variant into account?

Reply:
It should be possible to return the template for the current variant, right?

Reply:
The thing I'm having trouble figuring out is that the better_html gem receives only the component class and not the instance. And identifier is a class method.

I'm trying to understand if the code for better_html would need to be enhanced for components or if we can somehow figure out the variant without having the component's instance (only the class and the source for the template are available at that exact point in the code I linked).

Let me know if you have ideas on how to adequately solve this.

Reply:
@vinistock sounds like we might be due for some pairing on this one. Might you be up for some time next week?

Reply:
That makes sense. I'll shoot you an email.